pkg/resolver/search.go

```diff
@@ -77,77 +82,55 @@ func (s *SearchResult) Related() []SearchRelatedResult {
 	return r
 }
 
-//=====================
-
-var trimAND string = " AND "
-
-func (s *SearchResult) buildSearchQuery(ctx context.Context, count bool) (string, []interface{}) {
-	var selectClause, whereClause, limitClause, limitStr, query string
-	var args []interface{}
-	// SELECT uid, cluster, data FROM search.resources  WHERE lower(data->> 'kind') IN (lower('Pod')) AND lower(data->> 'cluster') IN (lower('local-cluster')) LIMIT 10000
-	selectClause = "SELECT uid, cluster, data FROM search.resources "
+func (s *SearchResult) buildSearchQuery(ctx context.Context, count bool) {
+	var limit int
+	var selectDs *goqu.SelectDataset
+	var whereDs []exp.Expression
+
+	// Example query: SELECT uid, cluster, data FROM search.resources  WHERE lower(data->> 'kind') IN
+	// (lower('Pod')) AND lower(data->> 'cluster') IN (lower('local-cluster')) LIMIT 10000
+	//FROM CLAUSE
+	schemaTable := goqu.S("search").Table("resources")
+	ds := goqu.From(schemaTable)
+	//SELECT CLAUSE
 	if count {
-		selectClause = "SELECT count(uid) FROM search.resources "
-	}
-
-	whereClause = " WHERE "
-
-	for i, filter := range s.input.Filters {
-		klog.Infof("Filters%d: %+v", i, *filter)
-		// TODO: Handle other column names like kind and namespace
-		if filter.Property == "cluster" {
-			whereClause = whereClause + filter.Property
-		} else {
-			// TODO: To be removed when indexer handles this as adding lower hurts index scans
-			whereClause = whereClause + "lower(data->> '" + filter.Property + "')"
-		}
-		var values []string
-
-		if len(filter.Values) > 1 {
-			for _, val := range filter.Values {
-				klog.Infof("Filter value: %s", *val)
-				values = append(values, strings.ToLower(*val))
-				//TODO: Here, assuming value is string. Check for other cases.
-				//TODO: Remove lower() conversion once data is correctly loaded from indexer
-				// "SELECT id FROM search.resources WHERE status = any($1)"
-				//SELECT id FROM search.resources WHERE status = ANY('{"Running", "Error"}');
-			}
-			whereClause = whereClause + "=any($" + strconv.Itoa(i+1) + ") AND "
-			args = append(args, pq.Array(values))
-		} else if len(filter.Values) == 1 {
-			whereClause = whereClause + "=$" + strconv.Itoa(i+1) + " AND "
-			val := filter.Values[0]
-			args = append(args, strings.ToLower(*val))
-		}
+		selectDs = ds.Select(goqu.COUNT("uid"))
+	} else {
+		selectDs = ds.Select("uid", "cluster", "data")
 	}
-	if s.input.Limit != nil {
-		limitStr = strconv.Itoa(*s.input.Limit)
+	//WHERE CLAUSE
+	if s.input != nil && len(s.input.Filters) > 0 {
+		whereDs = WhereClauseFilter(s.input)
 	}
-	if limitStr != "" {
-		limitClause = " LIMIT " + limitStr
-		query = selectClause + strings.TrimRight(whereClause, trimAND) + limitClause
-
+	//LIMIT CLAUSE
+	if s.input != nil && s.input.Limit != nil && *s.input.Limit != 0 {
```

Comment:
I **think** that the API will accept negative values, to be safe lets check limit is > 0.
```go
if s.input != nil && s.input.Limit != nil && *s.input.Limit > 0 {
```

Reply:
Also if `count == true` then a LIMIT doesn't apply.

Reply:
Done